widget/embed_ox_talks.js

```diff
@@ -1,116 +1,175 @@
+        element.empty();
+        for(var i=0; i<data._embedded.talks.length; i++) {
+            talk = data._embedded.talks[i];
+            header = $('<h3>');
+            var talk_title = talk.title.length>0 ? talk.title : "Title TBA";
+            header.html(talk_title);
+            element.append(header);
 
+            description = $('<p>');
+            description.html(talk.description);
+            element.append(description);
 
-//Performs the query then calling the specified callback function with the results and the selector
-var queryTalks = function (params, callback, selector) {
+            bullets = $('<ul>');
+            bullets.append($('<li>Speaker: ' + this.get_person_titles(talk._embedded.speakers) + '</li>'));
+            start_time = new Date(talk.start);
+            end_time = new Date(talk.end);
+    //        bullets.append($('<li>' + start_time.toLocaleDateString() 
+    //                         + ', ' + start_time.toLocaleTimeString() 
+    //                         + " - " + end_time.toLocaleTimeString() + '</li>'));
+            bullets.append($('<li>' + talk.formatted_date + '</li>'));
+            var location = talk._embedded.venue ? talk._embedded.venue.name : "Venue TBA";
+            bullets.append($('<li>' + location + '</li>'));
+            element.append(bullets);
+        }
+        if(data._links.prev) {
+            var prevLink = $('<a href="#">Prev</a>')
+            element.append(prevLink);
+            prevLink.on('click', (function(ev) {
+                var element = $(ev.target).parent();
+                this.queryURL(data._links.prev.href, this.buildList, element);
+            }).bind(this));
+        }
+        if(data._links.next) {
+            var nextLink = $('<a href="#">Next</a>');
+            element.append(nextLink);
+            nextLink.on('click', (function(ev) {
+                var element = $(ev.target).parent();
+                this.queryURL(data._links.next.href, this.buildList, element);
+            }).bind(this));
+        }
+    },
 
-    //NB. The response will always be via https: so the request should be made that way too to avoid a browser error.
-    var url_stem = "https://talks-dev.oucs.ox.ac.uk/api/events/search?";
-    var terms = [];
-    if (params.from) {
-        terms.push('from=' + params.from);
-    }
-    if (params.to) {
-        terms.push('to=' + params.to);
-    }
-    terms = terms.concat( buildTermsFromArray(params.speakers, 'speaker') );
-    terms = terms.concat( buildTermsFromArray(params.venues, 'venue') );
-    terms = terms.concat( buildTermsFromArray(params.organising_departments, 'organising_department') );
-    terms = terms.concat( buildTermsFromArray(params.topics, 'topic') );
+    //Performs the query then calling the specified callback function with the results and the selector
+    queryTalks: function (params, callback, element) {
+        var url_stem = "http://talks.local:8000/api/events/search?";
```

Comment:
If `talks.ox.ac.uk` doesn't provide any service on `/api` then that it is a very good idea indeed... However it still remains a "beta" API and might evolve depending on users needs.

@ahaith maybe still make it a parameter (with a default value) so that we can easily change it for dev purposes..